postgres/stack.yml

```
@@ -1,11 +1,19 @@
 # Use postgres/example user/password credentials
-version: '3.1'
+version: '3.9'
 
 services:
 
   db:
     image: postgres
     restart: always
+    # set shared memory limit when using docker-compose
+    shm_size: 128mb
+    # or set shared memory limit when deploy via swarm stack
+    #tmpfs:
+    # - /tmp:size=128mb
+    #volumes:
+    # - type: tmpfs
+    #   target: /dev/shm
```

This should be a bit less: :eyes:

```suggestion
    #tmpfs:
    #  - /dev/shm:size=128m
```

(the `128mb` with a `b` suffix causes errors, and if you want that to apply to `/dev/shm`, it has to be on `/dev/shm`, not just `/tmp` :sweat_smile:)

---

@tianon, thanks for the good comments.  /tmp target was an unfortunate typo.
btw, limitations in short syntax (/dev/shm:size=128m) does not work ([compose-spec/docs#tmpfs-long-syntax](https://github.com/compose-spec/compose-spec/blob/master/05-services.md#long-syntax-5)).

I checked that the `b` suffix does not cause errors. Specifying units of measurement for megabytes can be `m` or `mb`
([compose-spec/docs#specifying-byte-values](https://github.com/compose-spec/compose-spec/blob/master/spec.md#specifying-byte-values)).

---

So glad we have a second spec that's confusingly incompatible/inconsistent with the `docker run` flags it's wrapping :sweat_smile: :sob: :heart: